repo.go

```diff
@@ -103,6 +118,37 @@ func (r *Repo) Query(q string) (*Results, error) {
 	req.Header.Set("Content-Length", strconv.Itoa(len(b)))
 	req.Header.Set("Accept", "application/sparql-results+json")
 
+	return req, err
+}
+
+// Query performs a SPARQL HTTP request to the Repo, and returns the
+// parsed application/sparql-results+json response.
+//
+// The function accepts either a query in string format, or a struct
+// which fulfils the Provider interface and matches the GenericCall struct
+func (r *Repo) Query(queryProvider interface{}) (*Results, error) {
+
+	var req *http.Request
+	var err error
+
+	if p, ok := queryProvider.(Provider); ok {
+		req, err = p.GenRequest(r.endpoint)
+	}
+
+	if s, ok := queryProvider.(string); ok {
```

Comment:
The goal here is to maintain current behavior when passing a string to the `Query` function

Reply:
OK!